List ordered by amount of contributions and profile name

am2089 opened the pull request:
# What it Does
* Closes #105
* Presents the contributors list in the Contributor listview in order by amount of contributions and alphabetically.

# How I Tested
Built the app on simulator

am2089 committed:
List ordered by amount of contributions and profile name

mikaelacaron commented on Basic-Car-Maintenance/Shared/Settings/Views/ContributorsListView.swift:
```diff
@@ -14,7 +14,20 @@ struct ContributorsListView: View {
         List {
             if let contributors = viewModel.contributors, !contributors.isEmpty {
-                ForEach(contributors) { contributor in
+                let sortedContributors = contributors.sorted { (contributor1, contributor2) in
+                    
+                    // Sort by contributions (highest first), then by login name
+                    if contributor1.contributions > contributor2.contributions {
+                        return true
+                    } else if contributor1.contributions < contributor2.contributions {
+                        return false
+                    }
+                    
+                    // If contributions are equal, compare by login name
+                    return contributor1.login < contributor2.login
+                }
+                
+                ForEach(sortedContributors) { contributor in
```
Can you move this into the `SettingsViewModel` instead, you could do something similar to `sortedEvents` that's in the `DashboardViewModel`

Using a computed property

mikaelacaron commented on Basic-Car-Maintenance.xcodeproj/project.pbxproj:
```diff
@@ -801,6 +801,8 @@
 				"LD_RUNPATH_SEARCH_PATHS[sdk=macosx*]" = "@executable_path/../Frameworks";
 				MACOSX_DEPLOYMENT_TARGET = 13.4;
 				OTHER_LDFLAGS = "-ObjC";
+				"PRODUCT_BUNDLE_IDENTIFIER[sdk=iphoneos*]" = "com.andrewmuniz.Basic-Car-Maintenance";
+				"PRODUCT_BUNDLE_IDENTIFIER[sdk=macosx*]" = "com.andrewmuniz.Basic-Car-Maintenance";
```
Don't commit these! you can edit the `pbxproj` file manually in a text editor like VSCode (make sure Xcode is _not_ open when you do this

am2089 replied:
So you just want me to 'git ignore' this file next time I commit new changes to the branch so it does not push up with the changes?

mikaelacaron replied:
@am2089 **NO!!** the pbxproj must be there, this is basically what makes the xcodeproj file. You need to manually use a text editor and fix these parts that were added, because they shouldn't have been added

mikaelacaron commented on Basic-Car-Maintenance.xcodeproj/xcshareddata/xcschemes/Basic-Car-Maintenance-WidgetExtension.xcscheme:
```diff
@@ -90,7 +90,6 @@
       savedToolIdentifier = ""
       useCustomWorkingDirectory = "NO"
       debugDocumentVersioning = "YES"
```
Also fix this one!

am2089 replied:
Same question as the other file am I actually fixing anything or should I just  'git ignore ' this file since you said to fix it.

Thanks for all your help!

mikaelacaron replied:
make sure to add this back

mikaelacaron reviewed (changes requested):
A small change, for your actual issue, and then some issues with the `pbxproj` file, you're almost there! Thanks!

am2089 commented:
Ohhhh ok, got it Sent from my iPhoneOn Oct 10, 2023, at 10:50 PM, Mikaela Caron ***@***.***> wrote:﻿
@mikaelacaron commented on this pull request.



In Basic-Car-Maintenance.xcodeproj/project.pbxproj:
> +				"PRODUCT_BUNDLE_IDENTIFIER[sdk=iphoneos*]" = "com.andrewmuniz.Basic-Car-Maintenance";
+				"PRODUCT_BUNDLE_IDENTIFIER[sdk=macosx*]" = "com.andrewmuniz.Basic-Car-Maintenance";

@am2089 NO!! the pbxproj must be there, this is basically what makes the xcodeproj file. You need to manually use a text editor and fix these parts that were added, because they shouldn't have been added

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

am2089 committed:
Changes made topbxproj, and moved code from the contributorslistview to settingsviewmodel

am2089 commented:
I just pushed the changes to the branch, thanks again for being so patient!

On Tue, Oct 10, 2023 at 10:56 PM Andrew Muniz ***@***.***>
wrote:

> Ohhhh ok, got it
>
> Sent from my iPhone
>
> On Oct 10, 2023, at 10:50 PM, Mikaela Caron ***@***.***>
> wrote:
>
> ﻿
>
> ***@***.**** commented on this pull request.
> ------------------------------
>
> In Basic-Car-Maintenance.xcodeproj/project.pbxproj
> <https://github.com/mikaelacaron/Basic-Car-Maintenance/pull/127#discussion_r1353906426>
> :
>
> > +				"PRODUCT_BUNDLE_IDENTIFIER[sdk=iphoneos*]" = "com.andrewmuniz.Basic-Car-Maintenance";
> +				"PRODUCT_BUNDLE_IDENTIFIER[sdk=macosx*]" = "com.andrewmuniz.Basic-Car-Maintenance";
>
> @am2089 <https://github.com/am2089> *NO!!* the pbxproj must be there,
> this is basically what makes the xcodeproj file. You need to manually use a
> text editor and fix these parts that were added, because they shouldn't
> have been added
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mikaelacaron/Basic-Car-Maintenance/pull/127#discussion_r1353906426>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APDFTAWI3H6BLPM4G5SH7KDX6YCQTAVCNFSM6AAAAAA5ZVIQC6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTMNRZGY4TQMRUGA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
>

mikaelacaron commented on Basic-Car-Maintenance.xcodeproj/project.pbxproj:
```diff
@@ -101,7 +101,7 @@
 		FF0813562AD0A83000910EFA /* UITests.xcconfig */ = {isa = PBXFileReference; lastKnownFileType = text.xcconfig; path = UITests.xcconfig; sourceTree = "<group>"; };
 		FF0813572AD0A92700910EFA /* Widget.xcconfig */ = {isa = PBXFileReference; lastKnownFileType = text.xcconfig; path = Widget.xcconfig; sourceTree = "<group>"; };
-		FF098EFA2AB3424E003EC0FE /* Basic-Car-Maintenance.xcconfig */ = {isa = PBXFileReference; lastKnownFileType = text.xcconfig; path = "Basic-Car-Maintenance.xcconfig"; sourceTree = SOURCE_ROOT; };
+		FF098EFA2AB3424E003EC0FE /* Basic-Car-Maintenance.xcconfig */ = {isa = PBXFileReference; lastKnownFileType = text.xcconfig; name = "Basic-Car-Maintenance.xcconfig"; path = "Configurations/Basic-Car-Maintenance.xcconfig"; sourceTree = SOURCE_ROOT; };
```
It looks like you moved the config file, but you shouldn't have

mikaelacaron replied:
@am2089 Can you please fix this?

mikaelacaron commented on Basic-Car-Maintenance.xcodeproj/project.pbxproj:
```diff
@@ -840,6 +840,8 @@
 				"LD_RUNPATH_SEARCH_PATHS[sdk=macosx*]" = "@executable_path/../Frameworks";
 				MACOSX_DEPLOYMENT_TARGET = 13.4;
 				OTHER_LDFLAGS = "-ObjC";
+				"PRODUCT_BUNDLE_IDENTIFIER[sdk=iphoneos*]" = "com.andrewmuniz.Basic-Car-Maintenance";
+				"PRODUCT_BUNDLE_IDENTIFIER[sdk=macosx*]" = "com.andrewmuniz.Basic-Car-Maintenance";
```
Also be sure to remove this

mikaelacaron replied:
This hasn't been removed yet

mikaelacaron commented on Basic-Car-Maintenance/Shared/Settings/Views/ContributorsListView.swift:
```diff
@@ -8,18 +8,17 @@
-            if let contributors = viewModel.contributors, !contributors.isEmpty {
-                ForEach(contributors) { contributor in
+            if !viewModel.sortedContributors.isEmpty {
+                ForEach(viewModel.sortedContributors, id: \.id) { contributor in
```
you don't need the `id: \.id` because it conforms to `Identifiable`

mikaelacaron commented on Basic-Car-Maintenance/Shared/Settings/Views/ContributorsListView.swift:
```diff
@@ -8,18 +8,17 @@
-                        viewModel.urls["Basic-Car-Maintenance"]!) {
-                            ContributorsProfileView(name: contributor.login, url: contributor.avatarURL)
-                        }
+                            viewModel.urls["Basic-Car-Maintenance"]!) {
```
All the indentation is off
You can do `Cmd + A` to select everything and `Ctrl + i` to indent everything properly

mikaelacaron reviewed (changes requested):
Just a few small changes, and you're almost there! thanks!

am2089 committed:
changes requested

mikaelacaron reviewed (changes requested):
There's still several changes that are unresolved, please fix those and I can merge this, thanks! Afterwards click the re-review button
<img width="345" alt="Re-review button" src="https://github.com/mikaelacaron/Basic-Car-Maintenance/assets/22946902/65ebebdb-dfd3-427d-80b7-47c47aa66f28">

am2089 committed:
Changes

am2089 commented:
Ok so I moved the file to its original place lol

On Thu, Oct 12, 2023 at 7:49 PM Mikaela Caron ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> There's still several changes that are unresolved, please fix those and I
> can merge this, thanks! Afterwards click the re-review button
> [image: Re-review button]
> <https://user-images.githubusercontent.com/22946902/274757001-65ebebdb-dfd3-427d-80b7-47c47aa66f28.png>
> ------------------------------
>
> In Basic-Car-Maintenance.xcodeproj/project.pbxproj
> <https://github.com/mikaelacaron/Basic-Car-Maintenance/pull/127#discussion_r1357565197>
> :
>
> > @@ -101,7 +101,7 @@
>  		E58499692ACDDAFF00634660 /* Contributor.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = Contributor.swift; sourceTree = "<group>"; };
>  		FF0813562AD0A83000910EFA /* UITests.xcconfig */ = {isa = PBXFileReference; lastKnownFileType = text.xcconfig; path = UITests.xcconfig; sourceTree = "<group>"; };
>  		FF0813572AD0A92700910EFA /* Widget.xcconfig */ = {isa = PBXFileReference; lastKnownFileType = text.xcconfig; path = Widget.xcconfig; sourceTree = "<group>"; };
> -		FF098EFA2AB3424E003EC0FE /* Basic-Car-Maintenance.xcconfig */ = {isa = PBXFileReference; lastKnownFileType = text.xcconfig; path = "Basic-Car-Maintenance.xcconfig"; sourceTree = SOURCE_ROOT; };
> +		FF098EFA2AB3424E003EC0FE /* Basic-Car-Maintenance.xcconfig */ = {isa = PBXFileReference; lastKnownFileType = text.xcconfig; name = "Basic-Car-Maintenance.xcconfig"; path = "Configurations/Basic-Car-Maintenance.xcconfig"; sourceTree = SOURCE_ROOT; };
>
> @am2089 <https://github.com/am2089> Can you please fix this?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mikaelacaron/Basic-Car-Maintenance/pull/127#pullrequestreview-1675417811>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APDFTAQTUCJ75CMACGVDD4LX7B6Z7AVCNFSM6AAAAAA5ZVIQC6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTMNZVGQYTOOBRGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

mikaelacaron commented on Basic-Car-Maintenance.xcodeproj/project.pbxproj:
```diff
@@ -219,8 +219,8 @@
 			children = (
-				FFBE79BA2AD0A48C0005524E /* Configurations */,
 				FF098EFA2AB3424E003EC0FE /* Basic-Car-Maintenance.xcconfig */,
+				FFBE79BA2AD0A48C0005524E /* Configurations */,
```
Did you move the folder around? that's what it looks like here, you could manually fix this in a text editor

mikaelacaron reviewed (changes requested):
A few changes haven't been fixed yet

am2089 commented:
So I moved the folder to the original place from when we started the
project. and I deleted these two lines. Do you know where they would show
up? I made sure to delete them.

On Fri, Oct 13, 2023 at 10:57 PM Mikaela Caron ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> A few changes haven't been fixed yet
> ------------------------------
>
> In Basic-Car-Maintenance.xcodeproj/project.pbxproj
> <https://github.com/mikaelacaron/Basic-Car-Maintenance/pull/127#discussion_r1359079327>
> :
>
> >  				FF098EFA2AB3424E003EC0FE /* Basic-Car-Maintenance.xcconfig */,
> +				FFBE79BA2AD0A48C0005524E /* Configurations */,
>
> Did you move the folder around? that's what it looks like here, you could
> manually fix this in a text editor
> ------------------------------
>
> In Basic-Car-Maintenance.xcodeproj/project.pbxproj
> <https://github.com/mikaelacaron/Basic-Car-Maintenance/pull/127#discussion_r1359079261>
> :
>
> > +				"PRODUCT_BUNDLE_IDENTIFIER[sdk=iphoneos*]" = "com.andrewmuniz.Basic-Car-Maintenance";
> +				"PRODUCT_BUNDLE_IDENTIFIER[sdk=macosx*]" = "com.andrewmuniz.Basic-Car-Maintenance";
>
> This hasn't been removed yet
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mikaelacaron/Basic-Car-Maintenance/pull/127#pullrequestreview-1677692389>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APDFTARWW6E6SH5J2T72H2TX7H5RTAVCNFSM6AAAAAA5ZVIQC6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTMNZXGY4TEMZYHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

mikaelacaron commented:
> So I moved the folder to the original place from when we started the
project. and I deleted these two lines. Do you know where they would show
up? I made sure to delete them.

You can manually change it in a text editor the `pbxproj` is used with the `xcodeproj` for arranging files and folders. I'll look back at this and I can change it manually

am2089 commented:
Ohh okSent from my iPhoneOn Oct 13, 2023, at 11:21 PM, Mikaela Caron ***@***.***> wrote:﻿

So I moved the folder to the original place from when we started the
project. and I deleted these two lines. Do you know where they would show
up? I made sure to delete them.

You can manually change it in a text editor the pbxproj is used with the xcodeproj for arranging files and folders. I'll look back at this and I can change it manually

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

am2089 commented:
So is the PR accepted ?Sent from my iPhoneOn Oct 13, 2023, at 11:22 PM, Andrew Muniz ***@***.***> wrote:﻿Ohh okSent from my iPhoneOn Oct 13, 2023, at 11:21 PM, Mikaela Caron ***@***.***> wrote:﻿

So I moved the folder to the original place from when we started the
project. and I deleted these two lines. Do you know where they would show
up? I made sure to delete them.

You can manually change it in a text editor the pbxproj is used with the xcodeproj for arranging files and folders. I'll look back at this and I can change it manually

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

mikaelacaron committed:
Fix the pbxproj and xcscheme

mikaelacaron committed:
Fix pbxproj

mikaelacaron committed:
Fix WidgetExtension.xcscheme

mikaelacaron committed:
Fix WidgetExtension.xcscheme

mikaelacaron committed:
Fix xcscheme

mikaelacaron committed:
Remove WidgetExtension.xcscheme file

mikaelacaron reviewed (approved):
@am2089 See these latest changes. This is what I needed you to fix before this could be merged.
Fixing the `pbxproj` file and the `WidgetExtension.xcscheme` file

am2089 commented:
Hey, I have a question, were all these files messed up because I moved
files around from their original places? If there is another issue I want
to tackle, should I just re-fork the project so everything is already in
its place? Also thanks for being patient with me. It was my first pull
request ever lol.

On Sat, Oct 14, 2023 at 5:42 PM Mikaela Caron ***@***.***>
wrote:

> Merged #127
> <https://github.com/mikaelacaron/Basic-Car-Maintenance/pull/127> into dev.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mikaelacaron/Basic-Car-Maintenance/pull/127#event-10656226634>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APDFTAQZL4H6I3ISXKCRY23X7MWPZAVCNFSM6AAAAAA5ZVIQC6VHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJQGY2TMMRSGY3DGNA>
> .
> You are receiving this because you were mentioned.Message ID:
> <mikaelacaron/Basic-Car-Maintenance/pull/127/issue_event/10656226634@
> github.com>
>

mikaelacaron commented:
Hey!

> Hey, I have a question, were all these files messed up because I moved files around from their original places?

Yes that is part of it, any time you move a file, the `pbxproj` file will change (try it on a brand new clone of any iOS project, and you'll see what that looks like)


> If there is another issue I want to tackle, should I just re-fork the project so everything is already in its place?

Nope! Not needed, you will need to update your fork. This is a button on GitHub, see my livestream here for more details on exactly what this looks like (see the timestamp about git)

https://www.youtube.com/live/60ivxAOUrYU?si=4c6Eu92eKpndYrx3

> Also thanks for being patient with me. It was my first pull request ever lol.

You're welcome! That's okay, this is exactly what this project is for!! For others to learn and use git and GitHub in a real environment with other developers
